python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -166,7 +166,7 @@ def __init__(self,
                  transient: bool = False,
                  class_name: Optional[str] = None) -> None:
         self.rdd = rdd
-        self.user_caching = False
+        self.user_caching = self.rdd.is_cached
```

Comment:
It makes more sense that if the rdd for SparkXShards is already cached by user, we will always not uncache it?